File: backend/core/TransferObjects/SendMessage.cs
namespace core;

public class ClientSendsMessageToRoom: BaseTransferObject
{

    public string messageContent { get; set; }
    public int roomId { get; set; }
    
}


public class ServerBroadcastsMessageToClients : BaseTransferObject
{

    public Message message;
    
}

File: frontend/src/app/models/sendMessage.ts
import {Message} from "./entities";
import {BaseTransferObject} from "./baseTransferObject";

export class ClientWantsToLoadOlderMessages extends BaseTransferObject {
  roomId?: number;
  lastMessageId?: number;
  constructor(init?: Partial<ClientWantsToLoadOlderMessages>) {
    super();
    Object.assign(this, init);
  }
}

export class ServerSendsOlderMessagesToClient extends BaseTransferObject {
  roomId?: number;
  messages?: Message[];
  constructor(init?: Partial<ServerSendsOlderMessagesToClient>) {
    super();
    Object.assign(this, init);
  }
}
